### PR TITLE
Additional feat, chore, and fixes

### PR DESCRIPTION
**Pull request by jdalton**
Additional feat, chore, and fixes

### CODE REVIEW

**Comment by jdalton** [comment]
@steven-esser This sync will resolve all open non-dep-bot issues :)

**Review by steven-esser — APPROVED** [review]

**Comment by steven-esser** [comment]
@jdalton Good to Go!

I can go-ahead and push an npm release as well once these changes are in!

Thanks again for all the work.

**Comment by jdalton** [comment]
Thank you! I'll do follow-ups later in the next week or so with more tests too.

File: src/encode.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+function encodeSubpath(subpath) {
+    return isNonEmptyString(subpath)
+        ? encodeURIComponent(subpath).replace(/%2F/g, '/')
+        : ''
+}
+
+function encodeVersion(version) {
+    return isNonEmptyString(version)
+        ? encodeURIComponent(version).replace(/%3A/g, ':').replace(/%2B/g, '+')
```
Comment (Strum355) [inline src/encode.js:55]: This doesnt seem to be correct, Im not sure why some encodings are being reversed but from any source that I can read (between the spec and discussions/PRs in the spec repo e.g. https://github.com/package-url/purl-spec/issues/58#issuecomment-3009152400) versions are meant to be fully encoded @jdalton

Comment (jdalton) [reply src/encode.js:55]: Hi @Strum355 The code you're commenting on was committed on August 20, 2024. The comment you're referencing to spec clarifications was on June 26, 2025. I hope that clarifies the "why". Clarification was needed.

Comment (Strum355) [reply src/encode.js:55]: Thanks for pointing that out :smile: I've opened a quick PR to update this https://github.com/package-url/packageurl-js/pull/90

**Pull request merged**